hera_cal/lstbin.py

```diff
@@ -0,0 +1,893 @@
+            lst_low=None, lst_hi=None, wrap_point=2*np.pi, atol=1e-8, median=False,
+            sigma_clip=False, sigma=2.0, return_no_avg=False, verbose=True):
+    """
+    Bin data in Local Sidereal Time (LST) onto an LST grid. An LST grid
+    is defined as an array of points increasing in Local Sidereal Time, with each point marking
+    the center of the LST bin.
+
+    Parameters:
+    -----------
+    data_list : type=list, list of DataContainer dictionaries holding complex visibility data
+
+    lst_list : type=list, list of ndarrays holding LST stamps of each data dictionary in data_list
+    
+    lst_grid : type=ndarray, uniform 1D ndarray of LST grid to bin data into (marking center of bin).
+                        Note: ideally this should not start at 0, but at 0 + dlst / 2.
+
+    wgts_list : type=list, list of DataContainer dictionaries holding weights for each dict in data_list
+
+    lst_init : type=float, starting LST to initialize lst_grid at (if not fed). This marks the left-edge
+                        of the first LST bin.
+
+    dlst : type=float, delta-LST spacing for lst_grid.
+
+    lst_low : type=float, lower bound in lst_grid centers used for LST binning
+
+    lst_hi : type=float, upper bound in lst_grid centers used for LST binning
+
+    wrap_point : type=float, end point of a single day in LST
+
+    atol : type=float, absolute tolerance for comparing floats in lst_list to floats in lst_grid
+
+    median : type=boolean, if True use median for LST binning, else use mean
+
+    sigma_clip : type=boolean, if True, perform a sigma clipping algorithm of the LST bins
+        on the real and imag components separately. Warning: This considerably slows down the code.
+
+    sigma : type=float, input standard deviation to use for sigma clipping algorithm.
+
+    return_no_avg : type=boolean, if True, return binned but un-averaged data and wgts.
+
+    Output: (data_avg, wgts_avg, data_std, lst_bins, data_num)
+    -------
+    data_avg : dictionary of data having averaged the LST bins
+
+    wgts_avg : dictionary of data weights averaged in LST bins
+    
+    data_std : dictionary of data with real component holding std along real axis
+        and imag component holding std along imag axis
+
+    lst_bins : ndarray containing final lst grid of data
+
+    data_num : dictionary containing the number of data points averaged in each LST bin.
+
+    if return_no_avg:
+        Output: (data_bin, wgts_bin)
+        data_bin : dictionary with (ant1,ant2,pol) as keys and ndarrays holding
+            un-averaged complex visibilities in each LST bin as values. 
+    """
+    # construct lst_grid if not provided
+    if lst_grid is None:
+        # get dlst if not provided
+        if dlst is None:
+            dlst = np.median(np.diff(lst_list[0]))
+        # construct lst_grid
+        lst_grid = np.arange(lst_init, lst_init + wrap_point - dlst/2, dlst) + dlst/2
```

Comment:
Why does the default go from pi -> 3pi? we don't seem to be wrapping at the wrap point in this scheme. 0 to 2pi makes more sense.